### DISCUSSION TITLE
How can I terminate the uart_terminal in the sample code uart_service.py?

### DISCUSSION THREAD

**Discussion by W1nDsoul** [hbldh/bleak#1304] [Q&A] [answered]
I tried connecting to a Bluetooth device and sending commands and receiving responses successfully through the sample code. However, I can't automatically exit the terminal mode in the program after sending a command and receiving a response, without manually inputting ctrl+. Is there any way to stop it automatically after I input a command and receive a response?

Thanks

**Comment by dlech** [comment, marked as answer]
If you want the whole program to exit, you can cancel all tasks like the disconnect handler.

**Reply by W1nDsoul** [reply]
How can I add a disconnect handler to the code? It seems that the program is stuck in a loop.

**Reply by dlech** [reply]
The example code already has a disconnect handler.
